core/src/main/scala/org/apache/spark/sql/delta/GeneratedColumn.scala

```diff
@@ -252,12 +252,12 @@ object GeneratedColumn extends DeltaLogging with AnalysisHelper {
         case Project(exprs, _) =>
           exprs.flatMap {
             case Alias(expr, column) =>
-              expr.references.map {
-                case a: AttributeReference => a.name
-                case other =>
-                  // Should not happen since the columns should be resolved
-                  throw DeltaErrors.unexpectedAttributeReference(s"$other")
-              }.toSeq :+ column
+              // Get just the outer level attributes or struct extractors
+              expr.collect {
+                case ExtractBaseColumn(nameParts, dataType)
```

Comment:
If there is a nested struct such as `a: struct< b: struct<c: int>, d: int>`, and the generation expression is `a.b.c + 1`, will this return `a.b.c` and `b.c`?

Reply:
It should just get `a.b.c`

Reply:
`expr.collect` will traverse all of nodes. Right? Then `b.c` will match `!dataType.isInstanceOf[StructType]` as well?

Reply:
`ExtractBaseColumn` will only return when it encounters an `AttributeReference`, so `b.c` would never match `ExtractBaseColumn` and would never even get to the `!dataType.isInstanceOf[StructType]`. I can add another nested layer to the tests to verify